api/lanes.py

```diff
@@ -374,32 +377,167 @@ def lane_for_other_languages(_db, exclude_languages):
+        # We'll just ignore facets and return whatever we find.
+        if not use_materialized_works:
+            query = self.works()
+        else:
+            query = self.materialized_works()
+        if not query:
+            return []
 
-class RecommendationLane(Lane):
-    """A lane of works recommended for a particular work"""
+        return self.randomized_sample_works(query, use_min_size=True)
+
+
+class RelatedBooksLane(LicensePoolBasedLane):
+    """A lane of Works all related to the Work of a particular LicensePool
+
+    Sublanes currently include a SeriesLane and a RecommendationLane"""
+
+    DISPLAY_NAME = "Related Books"
+
+    def __init__(self, _db, license_pool, full_name, display_name=None,
+                 mock_api=None):
+        sublanes = self._get_sublanes(_db, license_pool, mock_api=mock_api)
+        if not sublanes:
+            edition = license_pool.presentation_edition
+            raise ValueError(
+                "No related books for %s by %s" % (edition.title, edition.author)
+            )
+        super(RelatedBooksLane, self).__init__(
+            _db, license_pool, full_name,
+            display_name=display_name, sublanes=sublanes,
+            invisible=True
+        )
+
+    def _get_sublanes(self, _db, license_pool, mock_api=None):
+        sublanes = []
+
+        # Create a recommendations sublane.
+        try:
+            lane_name = "Recommendations for %s by %s" % (
+                license_pool.work.title, license_pool.work.author
+            )
+            sublanes.append(RecommendationLane(
+                _db, license_pool, lane_name, mock_api=mock_api
+            ))
+        except ValueError, e:
+            # NoveList isn't configured.
+            pass
+
+        # Create a series sublane.
+        series = license_pool.presentation_edition.series
+        if series:
+            lane_name = SeriesLane.lane_name_from_series_title(series)
+            sublanes.append(SeriesLane(_db, license_pool, lane_name))
+
+        return sublanes
+
+    def apply_filters(self, qu, *args, **kwargs):
+        # This lane is composed entirely of sublanes and
+        # should only be used to create groups feeds.
+        return None
 
+
+class SeriesLane(LicensePoolBasedLane):
+    """A lane of Works in a series based on a particular LicensePool"""
+
+    DISPLAY_NAME = "Other Books in this Series"
+
+    def apply_filters(self, qu, work_model=Work, *args, **kwargs):
+        edition = self.license_pool.presentation_edition
+        series = edition.series
+        if not series:
+            return None
+        qu = self.only_show_ready_deliverable_works(qu, work_model)
+
+        # Aliasing Edition here allows this query to function
+        # regardless of work_model and existing joins.
+        work_edition = aliased(Edition)
+        qu = qu.join(work_edition).filter(
+            work_edition.series==series,
+            work_edition.id!=edition.id
+        )
+        return qu
+
```

Comment:
How are the series feeds sorted? I would expect them to be sorted by series_position and then by title. This should handle both the 'Harry Potter' case and the 'For Dummies' case.